src/http.rs

```diff
@@ -123,6 +124,19 @@ where
                 })
                 .context(CallRequestSnafu)?;
 
+            let (parts, body) = response.into_parts();
+            let mut data_buf: Option<Vec<u8>> = Some(Vec::new());
```

Comment:
Here I make the data_buf into an option and implement the suggest .take() method onto it and utilized the extend_from_slice method to avoid cloning.

I have opted to pass the entire data into the vector but realize this might not be desired, while attempting to use the previous approach of `if buf.len() >= 20` I found myself not using the `take()` method and instead opting to use either `drain()` or other vector slicing methods and continuing with the option with the remaining vector bits. I found this approach of passing the entire buf to the internal metrics struct easier, but now am concerned with performance.

@bruceg would you suggest using the `buf.len() >= 20` logic or truncating / not storing into the vector? I find this makes implementing the `take()` method not feasible.

Reply:
For clarity, you will need to _both_ check if the data buffer was not previously taken and check for the length after extending the size from the data. Note that `data_buf.take().unwrap()` is effectively just `buf` since this is already in a block where `data_buf` is some, and then assigning `data_buf = None` after using it.

Reply:
I have implemented a change below, originally I was going to do this:
```
let body = body.map_data(move |data| {
    if let Some(buf) = &mut data_buf {
        buf.extend_from_slice(&data);
        if buf.len() >= 20 && data_buf.is_some() {
            let buf = data_buf.take().unwrap();
            emit!(http_client::HttpBodyDataReceived { buf });
        }
    }
    data
```
But there was borrowing issues going on and I couldn't fix it / have the expertise to maneuver it.

Essentially rust is mad about my data_buf.is_some() call and the fact I use &mut data_buf. 

So I opted to implement the one in the most recent commit.
I have added the data_buf = None call which means I don't have to use the take() call. This also made me change the implementation of the internal event struct from a vector to a referenced vector.

Reply:
On second thought, I could go about it this way:

```
let mut data_buf: Option<Vec<u8>> = Some(Vec::new());
let body = body.map_data(move |data| {
    if let Some(buf) = &mut data_buf {
        buf.extend_from_slice(&data);
        if buf.len() >= 20 {
            emit!(http_client::HttpBodyDataReceived {
                buf: buf
            });
            data_buf = None;
        }
    }
    data
}).boxed();
```

Let me know if this is ok to implement @bruceg I've tested and seems to work ok with my vector.toml file and running with vdev

Reply:
You are right about the borrowing into the event needing lifetimes. It's one of the warts of structs containing references in Rust.